src/SoftRobots.Inverse/component/constraint/BarycentricCenterEffector.inl

```diff
@@ -111,28 +106,10 @@ void BarycentricCenterEffector<DataTypes>::computeBarycenter()
     const unsigned int nbp = m_state->getSize();
     Coord barycenter = Coord();
     for (unsigned int i=0; i<nbp; i++)
-    {
-        barycenter[0] += positions[i][0]/Real(nbp);
-        barycenter[1] += positions[i][1]/Real(nbp);
-        barycenter[2] += positions[i][2]/Real(nbp);
-    }
-    d_barycenter.setValue(barycenter);
-}
+        for(sofa::Size j=0; j<DataTypes::Coord::total_size; j++)
+            barycenter[j] += positions[i][j]/Real(nbp);
```

Comment:
My suggestion as a replacement of this function:

```cpp
template<class DataTypes>
void BarycentricCenterEffector<DataTypes>::computeBarycenter()
{
    const ReadAccessor<sofa::Data<VecCoord> > positions = m_state->readPositions();
    if (const sofa::Size nbp = m_state->getSize())
    {
        const Coord barycenter = std::accumulate(positions->begin(), positions->end(), Coord{}, std::plus<Coord>())  / nbp;
        d_barycenter.setValue(barycenter);
    }
    else
    {
        d_barycenter.setValue(Coord{});
        msg_error() << "Trying to compute a barycenter from an empty list of positions.";
    }
}
```

It adds error message and cleaner computation of the barycenter. (I have not checked that it works as before)